### DISCUSSION TITLE
Compatibility between JAVA Netcdf V4 library and TDS V5

### DISCUSSION THREAD

**Discussion by jzigna** [Unidata/netcdf-java#1157] [Questions and Answers]
Hello,

We have developed a JAVA component using the OpenDAP protocol with the NetCDF 4.6.3 library to request data on a THREDDS server in V4.

Do you know if we are going to face to any incompatibility if we upgrade only the TDS server in V5 ? We would like to upgrade the NetCDF libray in V5 of our compoent in a second step. 

Thanks for your replies.

Jean-Michel

**Comment by haileyajohnson** [comment]
Hello! Can you clarify what you mean by a Java component with the netCDF library? Do you mean a client that is requesting data via REST API, or are an application running alongside a THREDDS server? In general version 5.x of the TDS will not be compatible with version 4 components.

**Reply by jzigna** [reply]
The MOTU component runs on a Apache Tomcat server and can serve files  by delegating extraction to Thredds server with NCSS or OpenDap protocols (see https://github.com/clstoulouse/motu). We would like to update the TDS server with the latest version including the security update. I was wondering if MOTU is going to work or if we need to update the NetCDF JAVA library to make it work.
Thanks a lot for your quick reply, it will help us to schedule our work if necessary.

**Reply by haileyajohnson** [reply]
If you are using the REST API to talk to a THREDDS server, you will have to update your NCSS request paths: in 4.x, the path was `/ncss/` and in 5.x it is `/ncss/grid/` or `ncss/point/` depending on the type of data you are requesting, The OPeNDAP protocol is unchanged.

I'm still not sure how you're using netCDF-Java, so I don't know if you'll need to upgrade it. TDS 5 uses netcdf-java 5, but that doesn't mean the netcdf-java 4 library can't read data served by TDS 5.
